extensions/security/runtime/src/main/java/io/quarkus/security/runtime/interceptor/check/SupplierRolesAllowedCheck.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.security.runtime.interceptor.check;
+
+import java.lang.reflect.Method;
+import java.util.function.Supplier;
+
+import io.quarkus.security.ForbiddenException;
+import io.quarkus.security.UnauthorizedException;
+import io.quarkus.security.identity.SecurityIdentity;
+
+public class SupplierRolesAllowedCheck implements SecurityCheck {
+
+    private final Supplier<String[]> allowedRolesSupplier;
+
+    private volatile String[] allowedRoles;
```

Comment:
Minor nitpick: This doesn't need to be `volatile` unless I am missing something. It could just be `final`.

Reply:
NEvermind, read the code wrong :facepalm: